server/src/main/java/com/skillmagnet/User/UserController.java

```diff
@@ -107,4 +120,56 @@ public ResponseEntity<User> loginUser(@RequestBody User unAuthorizedUser) {
         userRepository.save(user);
         return new ResponseEntity<>(user, HttpStatus.OK);
     }
+    
+    /**
+     * This endpoint is to be used when a user "finishes" a lesson
+     * If requested lesson exists it's added to the users set of completed lessons
+     * 
+     * Tracking lessons completed allows us to be able to update course progress
+     * in enrollment objects which is also completed here automatically
+     * 
+     * @param uid - user that finished course
+     * @param lid - lesson they completed
+     * @return
+     *         Success: User Object & OK
+     *         Failure: null & NOT_FOUND
+     */
+    @PostMapping("/user/lesson/{uid}/{lid}")
+    public ResponseEntity<User> addLessonCompleted(@PathVariable("uid") int uid, @PathVariable("lid") int lid){
+        // If either lesson or user doesn't exist, return not found
+        User user = userRepository.findById(uid);
+        Optional<Lesson> lessonToAdd = lessonRepository.findById(lid);
+        if(lessonToAdd.isEmpty() || user == null){
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+
+        // Safe to assume that the Optional Contains a lesson at this point
+        // so we can use .get() knowing it will be a lesson added to the set
```

Comment:
I'd argue this comment can be removed :laughing: 

In all seriousness tho, optional is used to clean up things like this. Not sure about your VS Code setup, but generally a linter will give you warnings if a value has a chance to be null. But usually the linter is smart enough to see that if you did a `isEmpty()` check, then it will know your value can't be null and thus won't show any warnings (similar to how it won't show warnings for user being possibly null if you have `if (user == null)`. If linters are smart enough to figure this out, I'd expect a dev would too

Reply:
u right lol, removed it